core/lib/instrument.c

```diff
@@ -4881,6 +4881,20 @@ dr_raw_tls_cfree(uint offset, uint num_slots)
     return os_tls_cfree(offset, num_slots);
 }
 
+DR_API
+void
+dr_raw_write_tls_slot(reg_id_t *tls_register, ushort offset, void *value)
+{
+    d_r_set_tls(offset, value);
```

Comment:
This ignores tls_register and instead hardcodes DR's register.

Reply:
I wonder now what's better, parameterize the macros in os.c to take in a tls_register, or tie this here to DR's segment only. DR only prob. not a good idea?